src/skvhttpclient/test/ClientIntegrationTest.cpp

```diff
@@ -41,6 +41,62 @@ Client client;
 const std::string collectionName = "k23si_test_collection";
 const std::string schemaName = "test_schema";
 
+// Helper functions
+// Verify that two records are equal
+void verifyEqual(dto::SKVRecord& first, dto::SKVRecord& second) {
+    first.visitRemainingFields([&second] (const auto& field, auto val1) {
+        auto val2 = second.deserializeField<typename decltype(val1)::value_type>(field.name);
+        K2EXPECT(k2::log::httpclient, val1.has_value(), val2.has_value());
+        if (val1.has_value()) {
+            K2EXPECT(k2::log::httpclient, val1.value(), val2.value());
+        }
+    });
+}
+
+// Compare a query result with refernce records.
+void verifyEqual(std::vector<dto::SKVRecord> first, std::vector<dto::SKVRecord*> second) {
```

Comment:
By reference or const reference here? Why does the second vector have pointers to SKVRecords?

Reply:
Changed first one to reference. Used copy for second arg, otherwise I can't pass temporary created vector. I have to assign the vector to a variable and pass it it.